ublox_derive/src/output.rs

```diff
@@ -512,6 +543,17 @@ pub fn generate_code_to_extend_bitflags(bitflags: BitFlagsMacro) -> syn::Result<
         },
     };
 
+    let serde = quote! {
+        impl serde::Serialize for #name {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: serde::Serializer,
+            {
+                serializer.serialize_str(format!("{:?}", self).as_str())
```

Comment:
bitflags auto implement Debug, could just use their debug to serialize to resolve most of the flags

Reply:
You can just serialize the as the "repr_ty" here - 

```rust
    let serialize_fn = format_ident!("serialize_{}", repr_ty.to_token_stream().to_string()); // not sure if there is a better way to construct this method name
    let serde = quote! {
        impl serde::Serialize for #name {
            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
            where
                S: serde::Serializer,
            {
                serializer.#serialize_fn(self.bits())
            }
        }
    };
```

Reply:
but we won't get the naming tho, the bitflags generated debug matches and returns the name, bits would only give us 

```
    "rec_stat": 1,
```
vs
```
    "rec_stat": "LEAP_SEC",
```

Reply:
I agree its nice to have the names like that for if a human wants to read the file. But if a tool wants to ingest this data then it has to be aware of the specific naming conventions, like is it `LEAP_SEC`, `LeapSec` or `leap_second`? The tool here is using `LEAP_SEC` but that's not an offical conversion afaik, it's only meant for debugging. There is no ambiguity when returning it as a number. Presumably someone who uses this is familiar with the spec and can extract the values out of the bitfield if they want